core/pkg/membrane/membrane.go

```diff
@@ -348,20 +354,24 @@ func New(options *MembraneOptions) (*Membrane, error) {
 		}
 	}
 
-	if options.Pool == nil {
-		// Create new pool with defaults
-		minWorkersEnv := utils.GetEnv("MIN_WORKERS", "1")
-		minWorkers, err := strconv.Atoi(minWorkersEnv)
-		if err != nil || minWorkers < 0 {
-			return nil, fmt.Errorf("invalid MIN_WORKERS env var, expected non-negative integer value, got %v", minWorkersEnv)
-		}
+	minWorkersEnv := utils.GetEnv("MIN_WORKERS", "1")
+	minWorkers, err := strconv.Atoi(minWorkersEnv)
+	if err != nil || minWorkers < 0 {
+		return nil, fmt.Errorf("invalid MIN_WORKERS env var, expected non-negative integer value, got %v", minWorkersEnv)
+	}
 
-		maxWorkersEnv := utils.GetEnv("MAX_WORKERS", "100")
-		maxWorkers, err := strconv.Atoi(maxWorkersEnv)
-		if err != nil || minWorkers < 0 {
-			return nil, fmt.Errorf("invalid MAX_WORKERS env var, expected non-negative integer value, got %v", maxWorkersEnv)
-		}
+	maxWorkersEnv := utils.GetEnv("MAX_WORKERS", "300")
+	maxWorkers, err := strconv.Atoi(maxWorkersEnv)
+	if err != nil || minWorkers < 0 {
+		return nil, fmt.Errorf("invalid MAX_WORKERS env var, expected non-negative integer value, got %v", maxWorkersEnv)
+	}
+
+	if minWorkers > maxWorkers {
+		maxWorkers = minWorkers
+	}
```

Comment:
```go
maxWorkers = math.Max(minWorkers, maxWorkers)
```

Reply:
only works with floats, that is why I did it this way. Could convert as alternative